pycloudlib/azure/security_types.py

```diff
@@ -0,0 +1,106 @@
+"""Azure Security Types Classes."""
+# pylint: disable=W0107
+import abc
+from enum import Enum
+from typing import Any, Dict
+
+from pycloudlib import util
+
+
+class AzureCVMOSDiskEncryption(Enum):
+    """Represents Azure OS disk encryption types."""
+
+    VM_GUEST_STATE_ONLY = "VMGuestStateOnly"
+    DISK_WITH_VM_GUEST_STATE = "DiskWithVMGuestState"
+
+
+class SecurityType(abc.ABC):
```

Comment:
Nit: After removing `is_correct_vm_size` this class hierarchy / interface has only one method and a static attribute. Would not be simpler to encode this as a function? Like:

```py
def configure_security_vm_params(azure_security_type, vm_params, os_disk_enc):
    ...
```

Reply:
After our discussion, and abit of research, I see this will make sense for our current condition. Incase the security types will need more work, we can switch back to classes.